### DISCUSSION TITLE
Hyperlinked_grep and vim/neovim

### DISCUSSION THREAD

**Discussion by dpetka2001** [kovidgoyal/kitty#5526] [General]
Is there a way to open the hyperlinks from the result of hyperlinked_grep in an already existing vim/neovim 
opened instance? Currently it opens immediately in the current kitty window, so if you want to run hyperlinked_grep
again you must do so in a new window every time (since the window always gets consumed from clicking on the hyperlink).
I would like to be able to maintain only a single vim/neovim instance with open buffers from the hyperlinks if it is possible at all.

**Comment by kovidgoyal** [comment]
You can run any command you like when clicking a hyperlink, see https://sw.kovidgoyal.net/kitty/open_actions/

As for how to open it in an existing editor instance, that will depend
on whatever IPC support your editor provides. Figure out if the editor
has some way to run a command and have it open a file in it. And run
that command in your open-actions.conf

**Reply by trygveaa** [reply]
Vim supports this with the `--servername` argument. Start vim where you want the file opened with `vim --servername <server_name>` and configure the hyperlink command to be `vim --servername <server_name> --remote ${FILE_PATH}`.

Your vim will have to be compiled with `+clientserver`.

**Reply by dpetka2001** [reply]
I will try with --servername. Thank you for your help.

**Comment by dpetka2001** [comment]
So for anyone wondering about the same situation. Neovim does not support `+cmd` after `--remote`. It considers every argument given as part of the files you want to open. So for example, it would open 2 files (one named `+line_number` and the other `file_path`). In order to come around this problem I installed neovim-remote, which has as an executable `nvr`. So I configured the hyperlink command to use `nvr` instead of nvim and everything works like a charm. It opens the file at the exact line number provided by the fragment of hyperlinks at the Neovim instance that is already opened as a server. Again thank you all for the initial help you provided to me to solve this problem.
